be/src/formats/parquet/group_reader.cpp

```diff
@@ -110,38 +101,55 @@ Status GroupReader::get_next(ChunkPtr* chunk, size_t* row_count) {
     }
 
     if (has_filter) {
-        size_t hit_count = chunk_size >= 0 ? chunk_size : SIMD::count_nonzero(chunk_filter.data(), count);
+        size_t hit_count = chunk_size >= 0 ? chunk_size : SIMD::count_nonzero(chunk_filter.data(), rows_read);
         if (hit_count == 0) {
             active_chunk->set_num_rows(0);
-        } else if (hit_count != count) {
-            active_chunk->filter_range(chunk_filter, 0, count);
+        } else if (hit_count != rows_read) {
+            active_chunk->filter_range(chunk_filter, 0, rows_read);
         }
         active_chunk->check_or_die();
     }
 
+    Status status = Status::OK();
+    if (rows_read < rows_to_read) {
+        status = Status::EndOfFile("");
+    }
+
     size_t active_rows = active_chunk->num_rows();
     if (active_rows > 0 && !_lazy_column_indices.empty()) {
-        ChunkPtr lazy_chunk = _create_read_chunk(_lazy_column_indices);
-        RETURN_IF_ERROR(_lazy_skip_rows(_lazy_column_indices, lazy_chunk, *row_count));
-
         SCOPED_RAW_TIMER(&_param.stats->group_chunk_read_ns);
-        // read data into lazy chunk
-        _column_reader_opts.context->filter = has_filter ? &chunk_filter : nullptr;
-        Status st = _read(_lazy_column_indices, &count, &lazy_chunk);
-        if (!st.ok() && !st.is_end_of_file()) {
-            return st;
+        ChunkPtr lazy_chunk = _create_read_chunk(_lazy_column_indices);
+        // skip previous accumulate rows to skip first
+        if (_previous_rows_to_skip > 0) {
+            ASSIGN_OR_RETURN(size_t rows_skip, _skip(_lazy_column_indices, _previous_rows_to_skip));
+            DCHECK_EQ(rows_skip, _previous_rows_to_skip);
+            _previous_rows_to_skip = 0;
         }
-        if (has_filter) {
-            lazy_chunk->filter_range(chunk_filter, 0, lazy_chunk->num_rows());
+        for (size_t index = 0; index < chunk_filter.size();) {
```

Comment:
I have some feelings that batch way: "read them all them do filter later" would be more efficient.

I guess here you've done it in a increment way:
1. how many continusous rows should I read like X
2. then I just read X rows
right?

Reply:
Yes, your understanding is right.

Maybe it will be more efficient in a batch way, if the filter like `[1, 0, 1, 0, 1, 0]`, we will frequently call `skip()` function, just like the volcano model, will occur lots of virtual function call.

Reply:
I will change it

Reply:
Add an simple adaptive read.